src/access/wac.ts

```diff
@@ -0,0 +1,81 @@
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { internal_fetchAcl } from "../acl/acl.internal";
+import { getAgentAccess as getAgentAccessWac } from "../acl/agent";
+import { WebId, WithServerResourceInfo } from "../interfaces";
+import { internal_defaultFetchOptions } from "../resource/resource";
+import { Access } from "./universal";
+import { Access as AclAccess } from "../acl/acl";
+
+function getDefaultAccess(): Access {
+  return {} as Access;
+}
+
+function aclAccessToUniversal(access: AclAccess): Access {
+  const universalAccess = getDefaultAccess();
+  // In ACL, denying access to an actor is a notion that doesn't exist, so an
+  // access is either granted or not for a given mode.
+  // This creates a misalignment with the ACP notion of an access being granted,
+  // denied, or simply not mentionned. Here, we convert the boolean vision of
+  // ACL into the boolean or undefined vision of ACP.
+  for (const [mode, accessMode] of Object.entries(access)) {
+    if (accessMode) {
+      // There is no convenient way to type `mode` as the keys to the Access interface
+      // @ts-ignore
+      universalAccess[mode] = accessMode;
+    }
+  }
+  return universalAccess;
+}
+
+/**
+ * For a given Resource, look up its metadata, and read the Access permissions
+ * granted to the given Agent.
+ *
+ * Note that this only lists permissions granted to the given Agent individually,
+ * and will not exhaustively list modes the given Agent may have access to because
+ * they apply to everyone, or because they apply to the Agent through a group for
+ * instance.
+ *
+ * @param resource The URL of the Resource for which we want to list Access
+ * @param agent The Agent for which the Access is granted
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns True for Access modes granted to the Agent, False for Access modes
+ * denied to the Agent, and undefined otherwise.
```

Comment:
`undefined` is denied in WAC :)

```suggestion
 * denied to the Agent.
```

Reply:
Ah, and then this should also remove the mention of "deny" rather than of undefined, I suppose.

Reply:
Right, but since there's no notion of "deny" in ACL, either this becomes "True for Access modes granted to the Agent, and undefined otherwise.", or remains as it is (the idea being that this comment ends up applying at the public universal API level, rather than at the WAC level)

Reply:
Yes, the universal API would get its own TSDoc, so I'd make sure that the doc here applies to what this specific function actually does (i.e. the former option).

Reply:
2c0f5a827dfe3efa5f09caa9263a3fd9adee67b0